ios/FluentUI/Command Bar/CommandBar.swift

```diff
@@ -0,0 +1,290 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import UIKit
+
+/**
+ `CommandBar` is a horizontal scrollable list of icon buttons divided by groups.
+ Provide `itemGroups` in `init` to set the buttons in the scrollable area. Optional `leadingItem` and `trailingItem` add fixed buttons in leading and trailing positions. Each `CommandBarItem` will be represented as a button.
+ Set the `delegate` property to determine whether a button can be selected and deselected, and listen to selection changes.
+ */
+@objc(MSFCommandBar)
+open class CommandBar: UIView {
+    // Hierarchy:
+    //
+    // leadingButton
+    // containerView
+    // |--layer.mask -> containerMaskLayer (fill containerView)
+    // |--subviews
+    // |  |--scrollView (fill containerView)
+    // |  |  |--subviews
+    // |  |  |  |--stackView
+    // |  |  |  |  |--buttons (fill scrollView content)
+    // trailingButton
+
+    // MARK: - Public methods
+
+    public init(itemGroups: [CommandBarItemGroup], leadingItem: CommandBarItem? = nil, trailingItem: CommandBarItem? = nil) {
+        self.itemGroups = itemGroups
+
+        super.init(frame: .zero)
+
+        if let leadingItem = leadingItem {
+            self.leadingButton = button(forItem: leadingItem, isFixed: true)
+        }
+        if let trailingItem = trailingItem {
+            self.trailingButton = button(forItem: trailingItem, isFixed: true)
+        }
+
+        translatesAutoresizingMaskIntoConstraints = false
+
+        configureHierarchy()
+        updateButtonsState()
+    }
+
+    @available(*, unavailable)
+    public required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    /// Apply `isEnabled` and `isSelected` state from `CommandBarItem` to the buttons
+    @objc public func updateButtonsState() {
+        for button in itemsToButtonsMap.values {
+            button.updateState()
+        }
+    }
+
+    // MARK: Public properties
+
+    @objc public weak var delegate: CommandBarDelegate?
+
+    // MARK: Overrides
+
+    public override var intrinsicContentSize: CGSize {
+        .zero
+    }
+
+    public override func layoutSubviews() {
+        super.layoutSubviews()
+
+        containerMaskLayer.frame = containerView.bounds
+        updateShadow()
+    }
+
+    // MARK: - Private properties
+
+    private let itemGroups: [CommandBarItemGroup]
```

Comment:
what about dynamically adding and removing the items based on your selection?

Reply:
Is it a requirement for the initial version of CCB? It would be quite complicated to implement and currently there is no animation specs. For now, only the state of the buttons can be updated via `public func updateButtonsState()`.

Reply:
@rdeassis is this something that Fluent wants to track as a potential future enhancement?